autogluon/task/base/base_task.py

```diff
@@ -66,6 +70,7 @@ def run_fit(cls, train_fn, search_strategy, scheduler_options):
                        args=record_args)
         return results
 
+    @classmethod
     @abstractmethod
```

Comment:
Shouldn't @abstractmethod be removed if it is a @classmethod?

Reply:
It is not obligatory, they can stay together. This means that `fit` method is a `abstractmethod` so all children must implement it and on the other hand with `classmethod` you indicate that it children should implement it as a `classmethod`. See the reference : https://docs.python.org/3.6/library/abc.html#abc.abstractmethod

Reply:
I see, sorry I misread abstractmethod to mean staticmethod. Thanks for the explaination!